js/foundation.offcanvas.js

```diff
@@ -154,63 +165,52 @@ class OffCanvas {
    */
   open(event, trigger) {
     if (this.$element.hasClass('is-open') || this.isRevealed) { return; }
-    var _this = this,
-        $body = $(document.body);
+    var _this = this;
 
-    if (this.options.forceTop) {
-      $('body').scrollTop(0);
+    if (this.options.forceTo === 'top') {
+      window.scrollTo(0, 0);
+    } else if (this.options.forceTo === 'bottom') {
+      window.scrollTo(0,document.body.scrollHeight);
     }
-    // window.pageYOffset = 0;
-
-    // if (!this.options.forceTop) {
-    //   var scrollPos = parseInt(window.pageYOffset);
-    //   this.$element[0].style.transform = 'translate(0,' + scrollPos + 'px)';
-    //   if (this.$exiter.length) {
-    //     this.$exiter[0].style.transform = 'translate(0,' + scrollPos + 'px)';
-    //   }
-    // }
+
     /**
      * Fires when the off-canvas menu opens.
      * @event OffCanvas#opened
      */
-
-    var $wrapper = $('[data-off-canvas-wrapper]');
-    $wrapper.addClass('is-off-canvas-open is-open-'+ _this.options.position);
-
     _this.$element.addClass('is-open')
 
-      // if (_this.options.isSticky) {
-      //   _this._stick();
-      // }
-
     this.$triggers.attr('aria-expanded', 'true');
     this.$element.attr('aria-hidden', 'false')
         .trigger('opened.zf.offcanvas');
 
-    if (this.options.closeOnClick) {
-      this.$exiter.addClass('is-visible');
+    // If `contentScroll` is set to false, add class and disable scrolling on touch devices.
+    if (this.options.contentScroll == false) {
+      $('body').addClass('is-off-canvas-open').on('touchmove', this._stopScrolling);
+    }
+
+    // If we have an overlay lets make it visible.
+    if (this.options.contentOverlay) {
+      this.$overlay.addClass('is-visible');
+    }
+
+    // If we have close on click and an overlay add a `is-closable` class.
+    if (this.options.closeOnClick && this.options.contentOverlay) {
+      this.$overlay.addClass('is-closable');
     }
 
     if (trigger) {
       this.$lastTrigger = trigger;
     }
```

Comment:
This seems like it should be right at the top of the method - we're simply stashing away the argument passed in